deploy/buildtools/buildconfigfile.py

```diff
@@ -0,0 +1,155 @@
+        else:
+            launch_time = strftime("%Y-%m-%d--%H-%M-%S", gmtime())
+
+        self.args = args
+
+        global_build_configfile = None
+        with open(args.buildconfigfile, "r") as yaml_file:
+            global_build_configfile = yaml.safe_load(yaml_file)
+
+        # aws specific options
+        self.agfistoshare = global_build_configfile['agfis-to-share']
+        self.acctids_to_sharewith = global_build_configfile['share-with-accounts'].values()
+
+        # this is a list of actual builds to run
+        builds_to_run_list = global_build_configfile['builds']
+
+        build_recipes_configfile = None
+        with open(args.buildrecipesconfigfile, "r") as yaml_file:
+            build_recipes_configfile = yaml.safe_load(yaml_file)
+
+        build_farm_hosts_configfile = None
+        with open(args.buildfarmconfigfile, "r") as yaml_file:
+            build_farm_hosts_configfile = yaml.safe_load(yaml_file)
+
+        build_recipes = dict()
+        for section_name, section_dict in build_recipes_configfile.items():
+            if section_name in builds_to_run_list:
+                build_recipes[section_name] = BuildConfig(
+                    section_name,
+                    section_dict,
+                    build_farm_hosts_configfile,
+                    self,
+                    launch_time)
+
+        self.hwdb = RuntimeHWDB(args.hwdbconfigfile)
+
+        self.builds_list = build_recipes.values()
+        self.build_ip_set = set()
+
+    def setup(self):
+        """ Setup based on the types of buildfarmhosts """
+        for build in self.builds_list:
+            auto_create_bucket(build.s3_bucketname)
+
+        # check to see email notifications can be subscribed
+        get_snsname_arn()
+
+    def request_build_farm_hosts(self):
+        """ Launch an instance for the builds. Exits the program if an IP address is reused. """
+
+        def categorize(seq):
+            class_names = list(map(lambda x: x.build_farm_host_dispatcher.__class__.__name__, seq))
+            class_builds_zipped = zip(class_names, seq)
+
+            d = defaultdict(list)
+            for c, b in class_builds_zipped:
+                d[c].append(b)
+
+            return d
+
+        cat = categorize(self.builds_list)
+
+        for bhd_class, build_farm_hosts in cat.items():
+            # batch launching build farm hosts of similar types
+            getattr(import_module("buildtools.buildfarmhostdispatcher"),
+                bhd_class).request_build_farm_hosts(list(map(lambda x: x.build_farm_host_dispatcher, build_farm_hosts)))
+
+            for build in build_farm_hosts:
+                num_ips = len(self.build_ip_set)
+                ip = build.build_farm_host_dispatcher.get_build_farm_host_ip()
+                self.build_ip_set.add(ip)
+                if num_ips == len(self.build_ip_set):
+                    rootLogger.critical("ERROR: Duplicate {} IP used when launching instance".format(ip))
```

Comment:
I agree this would be useful, however we use the IP address of the machine to determine what to build/execute on it (this is baked in in many places). I think this feature is something to work on in the future.

Reply:
Can you point me to another place in the building flow where the uniqueness of the hostnames matters?

Reply:
This is an example from the `main` branch (and also applies in this branch). We get the build configuration from the global list based on the assigned IP.

https://github.com/firesim/firesim/blob/9d69e4e1380306f22ada6425b71515c17bbede93/deploy/buildtools/buildconfig.py#L159-L165

Reply:
I wonder how this intersects with supernode?  Are you supporting supernode continuing to work with this PR or is that out of scope?